File: rweb_application/src/main/resources/static/src/model/ImageEntity.ts
import Link from "./Link";
import TextPart from "./TextPart";

export default class ImageEntity {

    private _name: TextPart[];
    private _url: string;
    private _likes: number;
    private _nickname: string;
    private _liked: boolean;
    private _links: Link[];

    constructor(name: TextPart[], likes: number, url: string, nickname: string, liked: boolean, links: Link[]) {
        this._name = name;
        this._likes = likes;
        this._url = url;
        this._nickname = nickname;
        this._links = links;
        this._liked = liked;
    }

    get liked(): boolean {
        return this._liked;
    }

    get name(): TextPart[] {
        return this._name;
    }

    get likes(): number {
        return this._likes;
    }

    get url(): string {
        return this._url;
    }

    get nickname(): string {
        return this._nickname;
    }

    get links(): Link[] {
        return this._links;
    }
}

File: rweb_application/src/main/java/ua/danit/photogramm/rweb/model/ImageEntity.java
package ua.danit.photogramm.rweb.model;

import java.util.List;

import org.springframework.hateoas.ResourceSupport;

/**
 * Image entity presented single image in image feed.
 *
 * @author Andrey Minov
 */
public class ImageEntity extends ResourceSupport {
  private List<TextPart> name;
  private String url;
  private Integer likes;
  private String nickname;
  private boolean liked;

  /**
   * Instantiates a new Image entity.
   *
   * @param name     the name
   * @param url      the url
   * @param likes    the likes
   * @param nickname the nickname
   * @param liked    the liked
   */
  public ImageEntity(List<TextPart> name, String url, Integer likes, String nickname,
                     boolean liked) {
    this.name = name;
    this.url = url;
    this.likes = likes;
    this.nickname = nickname;
    this.liked = liked;
  }

  public List<TextPart> getName() {
    return name;
  }

  public String getUrl() {
    return url;
  }

  public Integer getLikes() {
    return likes;
  }

  public String getNickname() {
    return nickname;
  }

  public boolean isLiked() {
    return liked;
  }
}
